### PR TITLE
change the structure of thrift tornado transport class

### PR DESCRIPTION
**Pull request by junchaowu**

### CODE REVIEW

File: python/tchannel/thrift/transport/socket.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def __init__(self, tchannel, hostport, service_name):
             name.
         """
         super(TChannelTransport, self).__init__(
-            tchannel, hostport, service_name
+            hostport, service_name
```
Comment (abhinav) [inline python/tchannel/thrift/transport/socket.py]: Don't you need to record the tchannel object somewhere in this class?

Comment (junchaowu) [inline python/tchannel/thrift/transport/socket.py]: you are right.

File: python/tchannel/thrift/transport/tornado_base.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            "read() must not be called directly. Use readFrame()."
+        )
+
+    def recv_call(self):
+        raise NotImplementedError(
+            "recv_call() not supported for Tornado. Use readFrame()."
+        )
+
+    @gen.coroutine
+    def flush(self):
```
Comment (abhinav) [inline python/tchannel/thrift/transport/tornado_base.py]: This doesn't need `coroutine` the decorator.

File: python/tchannel/thrift/transport/tornado_base.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+
+class TChannelTornadoTransportBase(TChannelTransportBase):
+    """A Thrift Transport to send requests over an open TChannel connection.
+
+    Works with Thrift clients generated with the ``tornado`` option enabled.
+
+    .. code-block::
+
+        transport = TChannelTornadoTransport(
```
Comment (abhinav) [inline python/tchannel/thrift/transport/tornado_base.py]: Move the example back to the implementation. It's specific to that implementation.

File: python/tchannel/thrift/transport/socket.py (hunk 1 above)
Comment (breerly) [inline python/tchannel/thrift/transport/socket.py]: isn't he doing that on line 64?

Comment (abhinav) [inline python/tchannel/thrift/transport/socket.py]: He is now. :)

File: python/tchannel/thrift/transport/tornado_base.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            "read() must not be called directly. Use readFrame()."
+        )
+
+    def recv_call(self):
+        raise NotImplementedError(
+            "recv_call() not supported for Tornado. Use readFrame()."
+        )
+
+    def flush(self):
+        raise NotImplementedError("Must be implemented.")
```
Comment (abhinav) [inline python/tchannel/thrift/transport/tornado_base.py]: Implementations shouldn't have to care that a queue is being used to send back responses. Add a method implementations can call to send a response back.

**Pull request merged**